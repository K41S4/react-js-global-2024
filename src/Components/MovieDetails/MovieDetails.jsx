import styles from './MovieDetails.module.css';

export const MovieDetails = ({ movie }) => {
  return (
    <div className={styles.container}>
      <img src={movie.imageUrl} alt="" className={styles.image} />
      <div>
        <div className={styles.headerContainer}>
          <h2 className={styles.title}>{movie.title}</h2>
          <span className={styles.rating}>{movie.rating}</span>
        </div>
        <div className={styles.detailsContainer}>
          <div className={styles.releaseYear}>{movie.releaseYear}</div>
          <div className={styles.runtime}>{movie.runtime}</div>
        </div>
        <div className={styles.description}>{movie.description}</div>
      </div>
    </div>
  );
};
