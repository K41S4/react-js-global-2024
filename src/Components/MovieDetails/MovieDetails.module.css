.container {
  display: flex;
  background-color: #232323;
  color: white;
  padding: 20px;
  border-radius: 8px;
  max-width: 800px;
  margin: auto;
  font-family: 'Segoe UI', sans-serif;
  letter-spacing: 1px;
}

.image {
  border-radius: 8px;
  max-width: 300px;
  margin-right: 30px;
}

.headerContainer {
  display: flex;
  align-items: center;
}

.movieName {
  font-size: 36px;
  font-weight: 300;
  margin-top: 0;
  margin-bottom: 10px;
  margin-right: 20px;
  text-transform: uppercase;
}

.rating {
  font-size: 18px;
}

.releaseYear,
.duration {
  color: #f65261;
  font-size: 18px;
  margin-bottom: 5px;
}

.releaseYear {
  margin-right: 50px;
}

.detailsContainer {
  display: inline-flex;
}

.description {
  color: #787878;
  font-weight: 300;
  font-size: 20px;
  line-height: 24.38px;
  margin-top: 50px;
}

@media (max-width: 768px) {
  .container {
    flex-direction: column;
    align-items: center;
  }

  .image {
    margin-bottom: 20px;
  }
}
