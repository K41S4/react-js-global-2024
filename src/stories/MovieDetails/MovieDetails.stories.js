import { MovieDetails } from '../../Components/MovieDetails/MovieDetails';

export default {
  title: 'Example/MovieDetails',
  component: MovieDetails,
  parameters: {
    layout: 'centered',
    backgrounds: {
      default: 'dark',
    },
  },
  tags: ['autodocs'],
};

export const MovieDetailsDefault = {
  args: {
    imageUrl:
      'https://m.media-amazon.com/images/M/MV5BYTViYTE3ZGQtNDBlMC00ZTAyLTkyODMtZGRiZDg0MjA2YThkXkEyXkFqcGc@._V1_.jpg',
    title: 'Pulp Fiction',
    releaseYear: '1999',
    rating: '9.0',
    runtime: '2h 14min',
    description: 'Description',
  },
};
