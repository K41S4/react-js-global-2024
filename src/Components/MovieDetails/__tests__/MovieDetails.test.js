import { render } from '@testing-library/react';
import { MovieDetails } from '../MovieDetails';
import { useLoaderData } from 'react-router-dom';

jest.mock('react-router-dom', () => ({
  useLoaderData: jest.fn(),
}));

describe('MovieDetails', () => {
  test('renders all details', () => {
    const testData = {
      imageUrl: 'imageUrl',
      title: 'testName',
      releaseYear: 'testYear',
      rating: 'testRating',
      runtime: 'testDuration',
      description: 'testDescription',
    };
    useLoaderData.mockImplementation(() => testData);

    const { asFragment } = render(<MovieDetails />);

    expect(asFragment()).toMatchSnapshot();
  });
});
