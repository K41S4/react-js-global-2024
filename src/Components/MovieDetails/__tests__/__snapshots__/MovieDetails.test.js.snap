// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails renders all details 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <img
      alt=""
      class="image"
      src="imageUrl"
    />
    <div>
      <div
        class="headerContainer"
      >
        <h2
          class="movieName"
        >
          testName
        </h2>
        <span
          class="rating"
        >
          testRating
        </span>
      </div>
      <div
        class="detailsContainer"
      >
        <div
          class="releaseYear"
        >
          testYear
        </div>
        <div
          class="duration"
        >
          testDuration
        </div>
      </div>
      <div
        class="description"
      >
        testDescription
      </div>
    </div>
  </div>
</DocumentFragment>
`;
